test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.integration;
+
+import io.netty.util.ResourceLeakDetector;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+import io.netty.util.internal.logging.Slf4JLoggerFactory;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.KeyValueTableFactory;
+import io.pravega.client.admin.KeyValueTableInfo;
+import io.pravega.client.control.impl.Controller;
+import io.pravega.client.netty.impl.ConnectionFactory;
+import io.pravega.client.netty.impl.ConnectionFactoryImpl;
+import io.pravega.client.stream.mock.MockController;
+import io.pravega.client.tables.KeyValueTable;
+import io.pravega.client.tables.KeyValueTableClientConfiguration;
+import io.pravega.client.tables.KeyValueTableConfiguration;
+import io.pravega.client.tables.impl.KeyValueTableFactoryImpl;
+import io.pravega.client.tables.impl.KeyValueTableTestBase;
+import io.pravega.common.util.ByteArraySegment;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.contracts.StreamSegmentStore;
+import io.pravega.segmentstore.contracts.tables.TableStore;
+import io.pravega.segmentstore.server.host.handler.PravegaConnectionListener;
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.common.TestUtils;
+import java.time.Duration;
+import java.util.Collections;
+import lombok.val;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Integration test for {@link KeyValueTable}s using real Segment Store and connection.
+ * The only simulated component is the {@link Controller} which is provided via the {@link MockController}.
+ */
+public class KeyValueTableTest extends KeyValueTableTestBase {
+    private static final String ENDPOINT = "localhost";
+    private static final String SCOPE = "Scope";
+    private static final KeyValueTableConfiguration DEFAULT_CONFIG = KeyValueTableConfiguration.builder().partitionCount(5).build();
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private ResourceLeakDetector.Level originalLevel;
```

Comment:
@andreipaduroiu recently added a new way to do this which is probably better, let's be sure when the merge is resolved we are using that.

Reply:
Yes, this branch is still forked off `feature-key-value-tables` which hasn't been updated in a long time.

Reply:
Merged with master; this should be fixed now.